server/src/test/java/org/elasticsearch/index/mapper/ObjectMapperMergeTests.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.mapper;
+
+import com.google.common.collect.ImmutableMap;
+import org.elasticsearch.common.Explicit;
+import org.elasticsearch.common.joda.FormatDateTimeFormatter;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.mapper.FieldMapper.CopyTo;
+import org.elasticsearch.index.mapper.FieldMapper.MultiFields;
+import org.elasticsearch.index.mapper.TextFieldMapper.TextFieldType;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.AfterClass;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.elasticsearch.cluster.metadata.IndexMetaData.SETTING_VERSION_CREATED;
+import static org.elasticsearch.index.mapper.ObjectMapper.Dynamic;
+import static org.elasticsearch.index.mapper.ObjectMapper.Nested;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class ObjectMapperMergeTests extends ESTestCase {
```

Comment:
Did you create a new test class because the existing `ObjectMapperTests` extends `ESSingleNodeTestCase` and wanted a pure unit test? Nothing agains that, I'm just wondering if it otherwise would make sense to join both, even if these tests don't really need a mock node running, but this would keep testing code for "ObjectMapper" in one place. I don't have a final opininon on this, maybe you or others have some thoughts?

Reply:
Yes, I wanted a pure unit test. I preferred it for a couple of reasons:

* Unit tests are faster to boot, therefore we can create more test classes each one focused on a specific behavior. This one is focused on the merge operation. The downside of this is you never know if you have run enough test classes to check a behavior change. That's why, at the final end, I usually run tests from a test package.
* As it's faster to run, it please me as developer. (The latter one may not be a good reason ...)

That said, I'm totally open to join both test classes for consistency.

Reply:
I agree. Let me think about maybe making this test the "official" ObjectMapperTests then and rename the existing one that needs a node setup to make it easier to find this when other ObjectMapper-related unit tests are written. I think that would follow the conventions in this project better.

Reply:
I think we can leave this a separate test for now. Ideally `ObjectMapperTests` would be a simple unit test so we could put these new tests there, but this is certainly out of scope for this PR. We can reorgnanize existing tests also later.

Reply:
Nice, I can open a PR to do so next week.

Reply:
That would be great!

Reply:
Got it :wink: